### PR TITLE
Integration testing

### LINKED ISSUE #50: `App` **TESTS**
**Unit**
- [ ] components render
**Integration**
- [ ] click wedding card takes you to `WeddingDetails` and add photo list -> add a guest -> add photo -> submit photo
- [ ] can't add wedding if empty fields
- [ ] can delete a wedding

### LINKED ISSUE #66: API calls: is having `else throw error` and `catch error` redundant?

### LINKED ISSUE #121: Refactor: Redundancies with api calls error handling
I don't think we need both the if/else conditions and another .catch block
```
export const getWeddings = () => {
  return fetch("http://matrimania-backend.herokuapp.com/api/v1/weddings/weddings/")
  .then(response => {
    if(response.ok) {
      return response.json()
    } else {
      throw Error
    }
  })
  .catch(error => {
    return "No weddings found"
  })
}
```

can likely just be:
```
// export const getWeddings = () => {
//   return fetch("http://matrimania-backend.herokuapp.com/api/v1/weddings/weddings/")
//   .then(response => response.json())
//   .catch(error => {
//     return "No weddings found"
//   })
// }
```

### LINKED ISSUE #122: Feature: user feedback when try to submit empty add wedding form

### LINKED ISSUE #123: AddWeddingForm tests
- [ ] unit test render
- [ ] test form can't be submitted with empty fields

### PR DESCRIPTION
**Pull request by hannakim91**
# Pull Request

## What's this PR do? 

- [X] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Testing
- [ ] Styling
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description of changes?
- Integration tests with async calls starting from App to its different page views (and back): add wedding form -> back to main view; individual wedding page -> add photo list -> add a guest -> add photo -> submit photo; individual wedding page -> delete
- Unit tests for add wedding form render and preventing empty form submission
- Error handling for submitting Add Wedding form without mandatory fields filled
- Slight modifications to user feedback/alt tags
## Any background context you want to provide? 
n/a

## What are the relevant tickets? 
* Addresses
* Closes #123, Closes #122, Closes #121, Closes #50, Closes #66

## Screenshots
n/a

## Questions:
n/a

### CODE REVIEW

**Review by ericberglund117 — APPROVED** [review]
All these tests look great! so thorough and I struggle with integration testing, so it is nice to see how it's done properly

File: public/index.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <!DOCTYPE html>
 <html lang="en">
   <head>
+    <meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests">
```
Comment (Kristiannmiller) [inline public/index.html:4]: Hope this fixes our deploy issue!

File: src/__tests__/App.test.tsx [hunk 2]
```diff
@@ -1,27 +1,249 @@
...
+    await waitFor(() => {});
+    const deleteButton = screen.getByText("Delete Wedding");
+    userEvent.click(deleteButton);
+
+    await waitFor(() => {});
+    const noWeddingsImage = screen.getByAltText("No weddings in storage");
+    expect(noWeddingsImage).toBeInTheDocument();
+  });
+
+  it('can display an updated photo list for a wedding after guests and photos are added', async () => {
```
Comment (Kristiannmiller) [inline src/__tests__/App.test.tsx:149]: New record for longest test! Looks great!

File: src/apiCalls.tsx [hunk 3]
```diff
@@ -6,18 +6,12 @@ export const getWeddingInfo = (weddingId:number) => {
...
   return fetch("http://matrimania-backend.herokuapp.com/api/v1/weddings/weddings/")
-  .then(response => {
-    if(response.ok) {
-      return response.json()
-    } else {
-      throw Error
-    }
-  })
-  .catch(error => {
+  .then(response => response.json())
```
Comment (Kristiannmiller) [inline src/apiCalls.tsx:9]: Thanks for getting this resolved. Using that response is super helpful

**Review by Kristiannmiller — APPROVED** [review]
This looks great! Fingers crossed the meta tag fixes our deploy issue!

**Pull request merged**